compiler/damlc/daml-prim-src/GHC/Show/Text.daml

```diff
@@ -0,0 +1,127 @@
+-- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+
+-- #define SHOW_TEXT_CHR_NEWTYPE
+
+module GHC.Show.Text
+  ( showsPrecText
+  ) where
+
+#ifdef SHOW_TEXT_CHR_NEWTYPE
+import GHC.Base (map)
+#endif
+import GHC.Base ((.), (++), ($))
+import GHC.Classes (Eq (..), Ord (..), (&&))
+import GHC.CString (fromString)
+import GHC.Err (error)
+import GHC.Types (Bool, Int, Text, primitive)
+
+-- # Exported
+
+showsPrecText : Int -> Text -> ShowS
+showsPrecText _ x =
+    showString "\""
+  . showLitText x
+  . showString "\""
+
+-- # Primitives
+
+primExplode : Text -> [Text]
+primExplode = primitive @"BEExplodeText"
+
+primTextToCodePoints : Text -> [Int]
+primTextToCodePoints = primitive @"BETextToCodePoints"
+
+primIntToText : Int -> Text
+primIntToText = primitive @"BEToText"
+
+-- # Internal
+
+explode : Text -> [Chr]
+
+#ifdef SHOW_TEXT_CHR_NEWTYPE
```

Comment:
What's the purpose of this CPP directive / where would it be configured normally? Should we remove it and default to one or the other?

Reply:
there's a commented out `#define` for it at the top of the file, i used it to check that i was respecting my invariants while translating from the Haskell version, but since newtypes are not for free in Daml, i left it disabled for `daml-prim`. 

i realize it adds a bit of noise, but i'm not sure what's the best thing to do. i don't envision anyone enabling the flag (except in the unlikely case we do add zero-cost newtypes), but morally it should be the version with newtypes. Only for performance we want the other one, but the type synonym makes me wary of making it the only one.

Reply:
I propose to prefix the directive with one or two `_`  and to add `DEBUG` or something like that to make it clear it should not be use in prod.
```suggestion
#ifdef __DEBUG__SHOW_TEXT_CHR_NEWTYPE
```

Reply:
done, thank you for the suggestion